kiwmi/desktop/layer_shell.c

```diff
@@ -44,7 +42,9 @@ kiwmi_layer_commit_notify(struct wl_listener *listener, void *UNUSED(data))
 
     struct wlr_box old_geom = layer->geom;
 
-    arrange_layers(output);
+    if (layer->layer_surface->current.committed != 0) {
+        arrange_layers(output);
+    }
```

Comment:
Hmm, i have just noticed that this function’s diff is different in #50. Will have to re-check that once i have a cleaner/stash-able tree.

Reply:
Using the new approach seems to have some problems with damage. As the new way is going to get in anyway, let’s keep it the old way here. I’m too lazy to debug it.